ISISScientificComputing/autoreduce-frontend
# Rework resetting of values in the webapp AR-1168

### Summary of work
The webapp no longer uses multiple copies of the form to reset values. Now uses dataset fields on the input tags to record `data-initial` and `data-default`. 

Adds buttons to allow resetting individual fields to initial or default. Simplifies the code behind Reset all and removes JQuery usage.

- `data-initial`: keeps the value used in this reduction run, always taken from `run.run_variables.all()`

- `data-default`: keeps the default value stored in the current `reduce_vars.py`

### Demo of changed behaviour
![autoreduce-frontend-pull-13](https://user-images.githubusercontent.com/9135965/120494909-1fcd2600-c3b4-11eb-8be1-02f8d0af4412.gif)

### How to test your work
Tests should pass.

Run the web app locally and play around with the reset buttons. With VSCode tasks ([how to start instructions](https://github.com/ISISScientificComputing/autoreduce/wiki/Just-Use-VSCode-&-Docker))
1. Execute task `Run Autoreduce container`
2. Execute task `Migrate all`
2. Run and Debug the `Webapp Docker Attach` configuration
3. You'll have to make a `reduce.py` and `reduce_vars.py`, follow the instructions [here](https://github.com/ISISScientificComputing/autoreduce/wiki/PR-testing-quickstart#are-you-going-to-be-submitting-runs--reruns)
3. Click the `http://127.0.0.1:8000/` in the terminal (VSCode will forward the port if running on an SSH remote)

Without VSCode tasks:
Follow the [Local installation](https://github.com/ISISScientificComputing/autoreduce/wiki/Local-installation#installing-autoreduce) steps, and at the end uninstall with `pip uninstall -y autoreduce-frontend` to avoid any code interference.

### Additional comments
